### PR TITLE
Check log file actually exists before enabling Log File button

### LINKED ISSUE #8691: Disabling Log File button when not using a log file
Can we grey out the button for the log file when logging to the terminal with `--no-log-to-file`? It tries to use the current working directory.
no-log-to-file

### PR DESCRIPTION
**Pull request by Wedge009**
Resolves #8691

### CODE REVIEW

**Comment by soliton-** [comment]
The empty check seems useless then?

Another idea would be to actually make sure log_path_ is empty if not logging to file. Since it's initialized from `lg::get_log_file_path()` it might be better if that returned an empty string or a sensible path not what it does now.

**Comment by Wedge009** [comment]
I don't know if it's necessarily useless, that's why I left it in. But yes, I did see that `log_path_` is initialised from `get_log_file_path()`. At the very least this resolves the immediate and specific concern of #8691 (at least in my testing).

**Comment by soliton-** [comment]
My point is that the empty check surely was already intended to work as a guard against not logging to file and presumably worked at some point. However now it cannot work since `log_path_` will never be empty. So IMO a better fix would be to make `get_log_file_path` actually return an empty string when not logging to file instead of the nonsensical path it does currently. Checking for file existence seems more like a workaround.

**Comment by Wedge009** [comment]
I think generally if things operate independently, the better. So I don't see the harm in leaving the non-empty check as-is and I also don't think a file check is a 'work-around'. I would consider the potential improvement to `get_log_file_path()` a separate issue.

@Pentarctagon Looks like you were the last to work with `lg::get_log_file_path()` - do you have any thoughts on this?
https://github.com/wesnoth/wesnoth/blob/9548f4c7b5c95acaa802457420afb8ab6207fe3b/src/log.cpp#L280-L283

**Comment by Pentarctagon** [comment]
I'm assuming the issue here is that the log file path value ends up being just ".log" in this case, so a solution there would be to change it to:
`return output_file_path_.empty() ? "" : output_file_path_+lg::log_file_suffix;`

**Comment by Wedge009** [comment]
Ah, okay. Before your most recent change, the function was just returning `output_file_path_`, so with the suffix now being included the string was no longer empty and therefore the the Log File button was being incorrectly enabled as described in #8691.

Is the file-exist check considered 'expensive'? I think it makes sense to keep the check to avoid this otherwise brittle connection between the Log File GUI and the log code.

Regardless, I'll update the log code when I get home, hopefully that will satisfy soliton's concern.

**Comment by Pentarctagon** [comment]
A single check here in code that's not performance critical is unlikely to matter, but just as a general rule anything that needs to perform disk I/O will comparatively be much slower than a check that doesn't need to do that.

**Comment by soliton-** [comment]
> I think generally if things operate independently, the better. So I don't see the harm in leaving the non-empty check as-is and I also don't think a file check is a 'work-around'. I would consider the potential improvement to `get_log_file_path()` a separate issue.

You're right that for this functionality of actually opening the log file it makes sense to check it's there. With my workaround comment I was thinking about this as a check for not logging to file which may or may not mean that the log file was created (yet). (Though I think in the current implementation the log file is created at start anyway and not when writing the first log message or so.)

**Review by soliton- — APPROVED** [review]

File: src/log.cpp [hunk 1]
```diff
@@ -279,7 +279,7 @@ std::optional<bool> log_dir_writable()
 
 std::string get_log_file_path()
 {
-	return output_file_path_+lg::log_file_suffix;
+	return output_file_path_.empty() ? "" : output_file_path_+lg::log_file_suffix;
```
Comment (Vultraz) [inline src/log.cpp:282]: IMO this should be made to return an `optional<string>`.

Comment (Wedge009) [reply src/log.cpp:282]: I haven't used `optional` before but is this what you meant?

Comment (Wedge009) [reply src/log.cpp:282]: Uh, looks like the change broke a lot of checks.

Comment (Pentarctagon) [reply src/log.cpp:282]: I think it's fine to leave it as it is for now. Making it an optional just means everywhere it's used you need to either get the value assuming its value is present or provide some other default value or functionality if it's not.

Comment (Wedge009) [reply src/log.cpp:282]: 'Leave it as it is' - meaning before the change to optional?

Comment (Pentarctagon) [reply src/log.cpp:282]: Yeah. Using optional would be the more technically correct way to do it, but I don't think it really matters much either way here.

**Comment by Wedge009** [comment]
Aside from the remark about using `optional`, any other concerns before merging?

**Pull request merged**